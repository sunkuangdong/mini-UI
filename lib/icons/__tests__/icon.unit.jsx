import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import { mount } from 'enzyme'

describe('icon', () => {
    it('icon gitHubSvg', () => {
        const json = renderer.create(<Icon icon='gitHubSvg' />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('icon reactSvg', () => {
        const json = renderer.create(<Icon icon='reactSvg' />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('icon webpackSvg', () => {
        const json = renderer.create(<Icon icon='webpackSvg' />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('icons onClick', () => {
        // jest给我一个fn
        const fn = jest.fn()
        // const fn2 = jest.fn()
        // mount 相当于把Icon放到一个假想的页面
        const component = mount(<Icon icon='reactSvg' onClick={fn} />)
        // 如何点击？
        // 先找到svg元素 点击这个元素
        component.find('svg').simulate('click')
        // toBeCalled就是看fn是否被调用
        expect(fn).toBeCalled()
        // expect(fn2).toBeCalled() // 测试能否失败
    })
})