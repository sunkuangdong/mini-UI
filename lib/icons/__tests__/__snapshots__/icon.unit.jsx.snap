// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon icon gitHubSvg 1`] = `
<svg
  className="mini-icon"
>
  <use
    xlinkHref="#gitHubSvg"
  />
</svg>
`;

exports[`icon icon reactSvg 1`] = `
<svg
  className="mini-icon"
>
  <use
    xlinkHref="#reactSvg"
  />
</svg>
`;

exports[`icon icon webpackSvg 1`] = `
<svg
  className="mini-icon"
>
  <use
    xlinkHref="#webpackSvg"
  />
</svg>
`;

exports[`icon icons onClick 1`] = `2`;
